.github/workflows/update-ruby.yml

```diff
@@ -0,0 +1,49 @@
+name: Update Ruby definitions
+
+on:
+  workflow_dispatch:
+    inputs:
+      ruby_version:
+          description: 'Ruby version'
+          required: true
+          default: '2.7.2'
+      openssl_version:
+            description: 'OpenSSL version'
+            required: true
+            default: '3.0.14'
+
+  repository_dispatch:
+    types: [update-ruby]
+
+jobs:
+  update-ruby:
+    name: Update Ruby definitions
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Store Ruby version
+        run: |
+          echo "RUBY_VERSION=${{ github.event.client_payload.ruby_version || github.event.inputs.ruby_version }}" >> $GITHUB_ENV
+
+      - name: Store ABI version
+        run: echo "ABI_VERSION=$(echo ${{ env.RUBY_VERSION }} | cut -d '.' -f 1-2)" >> $GITHUB_ENV
+
+      - name: Store OpenSSL version
+        run: echo "OPENSSL_VERSION=${{ github.event.client_payload.openssl_version || github.event.inputs.openssl_version }}" >> $GITHUB_ENV
+
+      - name: Run script/update-cruby
+        run: |
+          curl -sL https://cache.ruby-lang.org/pub/ruby/${{ env.ABI_VERSION }}/ruby-${{ env.RUBY_VERSION }}.tar.gz
+          curl -sL https://www.openssl.org/source/openssl-${{ env.OPENSSL_VERSION }}.tar.gz
+          script/update-cruby ${{ env.RUBY_VERSION }} ${{ env.OPENSSL_VERSION }} .
+
+      - name: commit and push
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git add .
+          git commit -m "rake sync"
+          git push
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Because this is using the default GITHUB_TOKEN [it will not trigger other workflows](https://github.com/orgs/community/discussions/25702#discussioncomment-3248819) like https://github.com/rbenv/ruby-build/blob/master/.github/workflows/mirror.yml
That seems a problem, as that one should run.
It will also not trigger the ci workflow, which doesn't seem good.
That could result in the archive not being mirrored or having an incorrect checksum, etc.

A solution is to use a PAT for this.

Or instead of pushing to create a PR, I think that's safer and clearer what's going on.
That's what I'm using for setup-ruby:
https://github.com/ruby/ruby-builder/blob/master/.github/actions/create-pr-to-setup-ruby/action.yml

Reply:
Thanks, I use https://github.com/rbenv/ruby-build/blob/master/.github/workflows/release.yml#L19 for this.